packages/lesswrong/server/manualMigrations/2024-08-14-undraftPublicRevisions.ts

```diff
@@ -0,0 +1,33 @@
+import { getSqlClientOrThrow } from "../sql/sqlClient";
+import { registerMigration } from "./migrationUtils";
+import type { ITask } from "pg-promise";
+
+export const undraftPublicPostRevisions = async (db: SqlClient | ITask<{}>) => {
+  await db.none(`
+    UPDATE "Revisions" AS r
+    SET
+      "draft" = FALSE,
+      "version" = CASE WHEN LEFT("version", 1) = '0' THEN '1.0.0' ELSE "version" END
```

Comment:
Undraft all revisions that are the current revision for a published post, and set the version to 1.0.0 if the current version is less than 1